File: raccoon-ui/src/api/dictData.ts
import request from '@/utils/request'

export interface DictDataQueryRequest {
  page?: number
  size?: number
  dictTypeCode?: string
  dictLabel?: string
  status?: number
}

export interface DictDataRequest {
  id?: number
  dictTypeCode: string
  dictLabel: string
  dictValue: string
  dictExtend?: string
  status?: number
  sort?: number
  remark?: string
}

export interface DictDataResponse {
  id: number
  dictTypeCode: string
  dictLabel: string
  dictValue: string
  dictExtend: string
  status: number
  sort: number
  remark: string
  tenantId: number
  createBy: string
  createTime: string
  updateBy: string
  updateTime: string
}

export function getDictDataPage(data: DictDataQueryRequest) {
  return request({
    url: '/dict/data/page',
    method: 'post',
    data
  })
}

export function getDictDataById(id: number) {
  return request({
    url: `/dict/data/${id}`,
    method: 'get'
  })
}

export function addDictData(data: DictDataRequest) {
  return request({
    url: '/dict/data/add',
    method: 'post',
    data
  })
}

export function updateDictData(data: DictDataRequest) {
  return request({
    url: '/dict/data/update',
    method: 'post',
    data
  })
}

export function deleteDictData(id: number) {
  return request({
    url: '/dict/data/delete',
    method: 'post',
    params: { id }
  })
}

export function getDictDataByTypeCode(dictTypeCode: string) {
  return request({
    url: `/dict/data/${dictTypeCode}`,
    method: 'get'
  })
}

File: raccoon-cloud/raccoon-cloud-system/src/main/java/com/raccoon/cloud/system/model/DictData.java
package com.raccoon.cloud.system.model;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("sys_dict_data")
public class DictData implements Serializable {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    private String dictTypeCode;

    private String dictLabel;

    private String dictValue;

    private String dictExtend;

    private Integer status;

    private Integer sort;

    private String remark;

    private Long tenantId;

    private String createBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    private String updateBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer isDeleted;
}
